### PR TITLE
Fix validating of amount when sending a collectible

### LINKED ISSUE #1564: Can't Send a Collectible
**Describe the bug**

I am unable to send a collectible even though I have enough ETH to cover the gas

**Screenshots**

image

**To Reproduce**
_Steps to reproduce the behavior_

Attempt to send a collectible

**Expected behavior**

User should be able to send collectible if having enough ETH to cover txn fee

**Smartphone (please complete the following information):**
 - Device: any
 - OS: any
 - Version 0.2.16

-------------------------------------------------------------
_to be added after bug submission by internal support / PM_
**Severity**

- How critical is the impact of this bug on a user?
- Add stats if available on % of customers impacted
- Is this visible to all users?
- Is this tech debt?

### PR DESCRIPTION
**Pull request by danjm**
fixes #1564

Fixes an error with sending collectibles.

https://streamable.com/f4pbdw

### CODE REVIEW

File: app/components/Views/SendFlow/Confirm/index.js [hunk 1]
```diff
@@ -632,7 +632,7 @@ class Confirm extends PureComponent {
 				weiBalance = toWei(Number(selectedAsset.assetBalance));
 				weiInput = toWei(transaction.value);
 				errorMessage = weiBalance.gte(weiInput) ? undefined : strings('transaction.insufficient');
-			} else if (selectedAsset.isETH) {
+			} else if (selectedAsset.isETH || selectedAsset.tokenId) {
```
Comment (rickycodes) [inline app/components/Views/SendFlow/Confirm/index.js:635]: should we be checking for either or, or both?

or should we just be checking for `selectedAsset.tokenId`?

**Review by rickycodes — APPROVED** [review]

**Comment by rickycodes** [comment]
left a comment, but I still approved since this lgtm

File: app/components/Views/SendFlow/Confirm/index.js (hunk 1 above)
Comment (danjm) [reply app/components/Views/SendFlow/Confirm/index.js:635]: I think checking both is okay. When sending a collectible, we only need to validate that the user's balance exceeds the total gas, which is what `weiInput` will equal in the case of a collectible.

Comment (estebanmino) [reply app/components/Views/SendFlow/Confirm/index.js:635]: could you use something like https://github.com/MetaMask/metamask-mobile/blob/develop/app/components/UI/TransactionEditor/index.js#L331 ?

**Review by estebanmino — CHANGES_REQUESTED** [review]
Added a comment about https://github.com/MetaMask/metamask-mobile/pull/1565#discussion_r424096125 would be nice to have since users can send collectibles from dapps that we're not aware of

**Review by estebanmino — COMMENTED** [review]
that doesn't need to be a blocker TBH  I think it wasn't on this flow

File: app/components/Views/SendFlow/Confirm/index.js (hunk 1 above)
Comment (danjm) [reply app/components/Views/SendFlow/Confirm/index.js:635]: @estebanmino added in 0951e548

**Comment by danjm** [comment]
I have added commit 26e8f1ac to address an issue found by @omnat:

> Next button not working issue... It happened when I went back to collectible screen from the review screen. Wouldn’t go next from there

This now works: https://streamable.com/uukqza

**Comment by ibrahimtaveras00** [comment]
@danjm was part of the fix an update to txn history as well?

I'm seeing a no fee when there is a fee

image

This is different in relation to how iOS shows sent collectible 

Screen Shot 2020-05-13 at 12 14 29 PM

**Comment by danjm** [comment]
@ibrahimtaveras00 I don't think any change in this PR could cause that "No fee" message

I just did a send of a collectible and it didn't happen.

Screenshot from 2020-05-13 14-02-06

**Comment by danjm** [comment]
However, it could be an unintended consequence / bug introduced by this PR. I investigating what could cause that and how we might make consistent reproduction steps.

**Comment by ibrahimtaveras00** [comment]
Interesting, I couldn't reproduce the last two attempts 

1st attempt:
Screen Shot 2020-05-13 at 1 39 28 PM

2nd attempt: 
http://recordit.co/Nbw9CecFYc

Bug fixes for both issues look good on both OS's 👍 , QA Passed

**Review by ibrahimtaveras00 — APPROVED** [review]

**Comment by estebanmino** [comment]
I think that could come from https://github.com/MetaMask/metamask-mobile/pull/1465/files#diff-9bf1465668b5085c8f828241d9dd6bbcR96 @danjm WDYT?

**Comment by danjm** [comment]
@estebanmino yeah, that's where the `undefined` fee is affecting the UI

The question is why is the fee undefined? I have not been able to track that down yet. Getting closer... I at least have a transaction with the fee undefined in state now, so I can inspect it.

**Comment by danjm** [comment]
@ibrahimtaveras00 Those last two examples you shared, the "Amount" is actually the collectible id number. Is that what the design is supposed to be?

**Comment by ibrahimtaveras00** [comment]
That's how the current implementation is on playstore and testflight versions and how it's been for quite some time.  I created https://github.com/MetaMask/metamask-mobile/issues/1563 to address this, but it's not something new from this release as it's been around.

CC: @cjeria

**Comment by danjm** [comment]
@ibrahimtaveras00 @estebanmino I figured out why sometimes we are seeing "No fee" and sometimes not. Different collectibles have different transfer methods. We encode some of these specifically in `/metamask-mobile/app/util/collectibles-transfer.json`. The data we attach to the transaction details is currently a little different depending on whether the collectible contract uses a  `transfer` or `transferFrom` method.

The fix is to ensure the fee data is included in both types of transaction data. This issue is separate from this PR. So I will create another PR for it.

**Review by estebanmino — APPROVED** [review]

**Pull request merged**